packages/features/ee/dsync/lib/handleUserEvents.ts

```diff
@@ -0,0 +1,60 @@
+import type { DirectorySyncEvent, User } from "@boxyhq/saml-jackson";
+
+import removeUserFromOrg from "@calcom/features/ee/dsync/lib/removeUserFromOrg";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import prisma from "@calcom/prisma";
+import { getTeamOrThrow } from "@calcom/trpc/server/routers/viewer/teams/inviteMember/utils";
+import type { UserWithMembership } from "@calcom/trpc/server/routers/viewer/teams/inviteMember/utils";
+
+import createUserAndInviteToOrg from "./users/createUserAndInviteToOrg";
+import dSyncUserSelect from "./users/dSyncUserSelect";
+import inviteExistingUserToOrg from "./users/inviteExistingUserToOrg";
+
+const handleUserEvents = async (event: DirectorySyncEvent, orgId: number) => {
+  const eventData = event.data as User;
+  const userEmail = eventData.email;
+  // Check if user exists in DB
+  const user = await prisma.user.findFirst({
+    where: {
+      email: userEmail,
+    },
+    select: dSyncUserSelect,
+  });
+
+  // User is already a part of that org
+  if (user?.organizationId && eventData.active) {
+    return;
```

Comment:
We probably need more granual information here - this can have a few different causes (user not found, event not active, user already in another organisation)

Reply:
So this is interesting. The person doing the syncing won't actually see any error messages since it should be done automatically. If it does throw an error then the directory platform will just say an error occured with the user. We could send an email to the org owner/admin.